### PR TITLE
Feature/post problem

### LINKED ISSUE #21: Create pop-up form for creating new problem (for problem name and grade) - this sends post request

### PR DESCRIPTION
**Pull request by jalbe0076**
### Summary: This PR will close #21 
- Describe what you did
  - Set up post request to mock server
  - Connected the current wall to be displayed when you try to create a new problem
  - Created a modal to enter wall name and grade, both are required fields, grade defaults to V0 but not entering a name does not save the problem and displays a prompt to a user
  - Updated info modal
  - Conditionally render menu options
  - Set up network error state and display messages to user if something goes wrong
- Any instructions for the reviewer?
  - Run `npm install` to install dependencies 
  - This get's the object from the mock server when logging it but this functionality will need to be tested when the back end is up and running. 
### Did this break anything?
- [X] No
- [ ]  Yes
### Type of change
- [X] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [X]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
### Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [X]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [X]  The code is DRY. If it's not, I tried my best
- [X]  I reviewed my code before pushing

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -6,34 +6,75 @@ import CreateProblem from "./app/Pages/CreateProblem";
 import Home from "./app/Pages/Home";
 import MenuBar from "./app/Components/MenuBar";
 import RouteContext from "./app/Components/RouteContext";
-import { NavigationContainer } from "@react-navigation/native";
+import { NavigationContainer } from '@react-navigation/native';
 import { createNativeStackNavigator } from "@react-navigation/native-stack";
 import ViewAllProblems from "../klime_fe/app/Pages/ViewAllProblems";
 import { GestureHandlerRootView } from "react-native-gesture-handler";
+import { postProblem } from "./app/Components/apiCalls";
+import Toast  from 'react-native-toast-message';
```
Comment (Jade-ZS) [inline App.js:14]: MP: seems that there is an extra space

File: app/Components/CreateProblemModal.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    { label: "V7", value: "V7" },
+    { label: "V8", value: "V8" },
+    { label: "V9", value: "V9" },
+    { label: "V10", value: "V10" },
+    { label: "V11", value: "V11" },
+    { label: "V12", value: "V12" },
+    { label: "V13", value: "V13" },
+    { label: "V14", value: "V14" },
+    { label: "V15", value: "V15" },
+];
```
Comment (Jade-ZS) [inline app/Components/CreateProblemModal.js:31]: MP: maybe a for loop will make this drier?

File: App.js [hunk 3]
```diff
@@ -6,34 +6,75 @@ import CreateProblem from "./app/Pages/CreateProblem";
...
+  const handleNetworkErrors = (error) => {
+    setNetworkErrors(error.message);
+  }
+
+  const showToast = (name) => {
+    Toast.show({
+      position: 'bottom',
+      text1: `Saved ${name}`,
+    });
+  }
```
Comment (Jade-ZS) [inline App.js:34]: I love this toast!

**Review by Jade-ZS — APPROVED** [review]
Thanks for the great work Jason! I'm with Hollis, absolutely love the toast too! I commented some minor points. Just to share my thoughts, don't feel pressure to follow and change anything 🤗

File: app/Components/CreateProblemModal.js (hunk 2 above)
Comment (jalbe0076) [reply app/Components/CreateProblemModal.js:31]: It would... I could take care of it in another branch.

**Pull request merged**